L1Trigger/Phase2L1ParticleFlow/src/l1-converters/tkinput_ref.cpp

```diff
@@ -0,0 +1,676 @@
+    : TrackInputEmulator(parseRegion(iConfig.getParameter<std::string>("region")),
+                         parseEncoding(iConfig.getParameter<std::string>("trackWordEncoding")),
+                         iConfig.getParameter<bool>("bitwiseAccurate")) {
+  if (region_ != Region::Endcap && region_ != Region::Barrel) {
+    edm::LogError("TrackInputEmulator") << "region '" << iConfig.getParameter<std::string>("region")
+                                        << "' is not yet supported";
+  }
+  debug_ = iConfig.getUntrackedParameter<bool>("debug", false);
+  configPt(iConfig.getParameter<uint32_t>("ptLUTBits"));
+  configEta(iConfig.getParameter<uint32_t>("etaLUTBits"),
+            iConfig.getParameter<int32_t>("etaPreOffs"),
+            iConfig.getParameter<uint32_t>("etaShift"),
+            iConfig.getParameter<int32_t>("etaPostOffs"),
+            iConfig.getParameter<bool>("etaSigned"),
+            region_ == Region::Endcap);
+  configPhi(iConfig.getParameter<uint32_t>("phiBits"));
+  configZ0(iConfig.getParameter<uint32_t>("z0Bits"));
+  if (region_ == Region::Barrel) {
+    configDEtaBarrel(iConfig.getParameter<uint32_t>("dEtaBarrelBits"),
+                     iConfig.getParameter<uint32_t>("dEtaBarrelZ0PreShift"),
+                     iConfig.getParameter<uint32_t>("dEtaBarrelZ0PostShift"),
+                     iConfig.getParameter<double>("dEtaBarrelFloatOffs"));
+    configDPhiBarrel(iConfig.getParameter<uint32_t>("dPhiBarrelBits"),
+                     iConfig.getParameter<uint32_t>("dPhiBarrelRInvPreShift"),
+                     iConfig.getParameter<uint32_t>("dPhiBarrelRInvPostShift"),
+                     iConfig.getParameter<double>("dPhiBarrelFloatOffs"));
+  }
+  if (region_ == Region::Endcap) {
+    configDEtaHGCal(iConfig.getParameter<uint32_t>("dEtaHGCalBits"),
+                    iConfig.getParameter<uint32_t>("dEtaHGCalZ0PreShift"),
+                    iConfig.getParameter<uint32_t>("dEtaHGCalRInvPreShift"),
+                    iConfig.getParameter<uint32_t>("dEtaHGCalLUTBits"),
+                    iConfig.getParameter<uint32_t>("dEtaHGCalLUTShift"),
+                    iConfig.getParameter<double>("dEtaHGCalFloatOffs"));
+    configDPhiHGCal(iConfig.getParameter<uint32_t>("dPhiHGCalBits"),
+                    iConfig.getParameter<uint32_t>("dPhiHGCalZ0PreShift"),
+                    iConfig.getParameter<uint32_t>("dPhiHGCalZ0PostShift"),
+                    iConfig.getParameter<uint32_t>("dPhiHGCalRInvShift"),
+                    iConfig.getParameter<uint32_t>("dPhiHGCalTanlInvShift"),
+                    iConfig.getParameter<uint32_t>("dPhiHGCalTanlLUTBits"),
+                    iConfig.getParameter<double>("dPhiHGCalFloatOffs"));
+  }
+}
+
+#endif
+
+l1ct::TrackInputEmulator::TrackInputEmulator(Region region, Encoding encoding, bool bitwise)
+    : region_(region),
+      encoding_(encoding),
+      bitwise_(bitwise),
+      rInvToPt_(31199.5),
```

Comment:
If  I understand correctly what happens with your three constructors here you always end up in using this one. So why do you need to have these constants as class members?

Reply:
@gpetruc Can you answer this question?

Reply:
in other standalone code using these classes it is useful to be able to vary them (and there interface methods to do it)